### DISCUSSION TITLE
in FeaturePlot, when choosing a slot, which assay in the Seurat object is used ?

### DISCUSSION THREAD

**Discussion by thugib** [satijalab/seurat#5005] [Q&A] [answered]
I have a Seurat object in which I have used SCTransform and then integrated the data. The resulting Seurat object has three assays; 
'RNA', 'SCT' and 'integrated'. Each of the three assays has slots for 'counts', 'data' and 'scale.data'. However, in the 'RNA' assay the 'scale.data' is empty (unpopulated, no numbers) and in the 'integrated' assay the 'counts' slot is empty. 

 When using FeaturePlot, I do not want to use integrated data, but FeaturePlot has no argument for choosing the assay. FeaturePlot does have an argument for choosing the slot in which the value can be either 'counts', 'data' or 'scale.data'. 
  
  When I choose 'counts' as the slot, then all the genes (features) come up as not being found, even when they are found when 'data' or 'scale.data' is the chosen slot. So, it seems like FeaturePlot is using the 'integrated' assay, but I want to avoid integrated data for plotting.

**Comment by timoast** [comment, marked as answer]
The default assay is used. This is the default for all functions in Seurat unless an `assay` parameter is implemented and a different assay is specified in the function call.

**Comment by thugib** [comment]
So, after integrating the data, I guess the 'integrated' assay is the default assay in which case to to differential expression or graphics I would need to do: `DefaultAssay(mySeuratObject) <- "RNA"`
and now the 'RNA' assay is the default. 

By the way, I am using Chrome as my browser and nothing happens when I click "Mark as answer". Also, when I click the "Got it!" button the pop-up does not go away, so maybe a browser problem with Chrome ?  I have tried a number of times on more than one question to 'Mark as answer', but it never seems to mark the question as answered.

**Reply by timoast** [reply]
You can find out what the default assay is by printing the object summary, or using the `DefaultAssay()` function

**Reply by thugib** [reply]
OK, great.  I was wondering if anywhere you may have a vignette style web page that would provide a small description of all the things you find in a Seruat object; a sort of anatomy of a Seurat object. For example, I would like to know the different between the 'data' slot and the 'counts' slot, or what exactly is the data for @i or @p ?

**Reply by timoast** [reply]
This gives an overview of the object structure: https://github.com/satijalab/seurat/wiki

You should not access slots via the `@` operator, we have functions defined to access all the data in available in the object. The `i` and `p` slots you're referring to are part of the sparse matrix object defined in the Matrix package.

**Comment by thugib** [comment]
Thank you very much for the link and all the help !
